File: DiKErnel/Src/DiKErnel.KernelWrapper.Json.Output/Internal/Data/JsonOutputLocationData.cpp
// Copyright (C) Stichting Deltares 2020. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, version 3.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
//
// All names, logos, and references to "Deltares" are registered trademarks of
// Stichting Deltares and remain full property of Stichting Deltares at all times.
// All rights reserved.

#include "JsonOutputLocationData.h"

#include <utility>

namespace DiKErnel::KernelWrapper::Json::Output
{
    using namespace std;

    JsonOutputLocationData::JsonOutputLocationData(
        string name,
        vector<double> damages,
        const int* timeOfFailure)
        : _name(move(name)),
          _damages(move(damages)),
          _timeOfFailure(timeOfFailure) {}

    const string& JsonOutputLocationData::GetName() const
    {
        return _name;
    }

    const vector<double>& JsonOutputLocationData::GetDamages() const
    {
        return _damages;
    }

    bool JsonOutputLocationData::GetLocationFailed() const
    {
        return _timeOfFailure != nullptr;
    }

    const int* JsonOutputLocationData::GetTimeOfFailure() const
    {
        return _timeOfFailure;
    }
}

File: Src/DiKErnel.KernelWrapper.Json.Output/Data/Generic/JsonOutputLocationData.cs
// Copyright (C) Stichting Deltares and State of the Netherlands 2023. All rights reserved.
//
// This file is part of DiKErnel.
//
// This program is free software; you can redistribute it and/or modify it under the terms of
// the GNU Lesser General Public License as published by the Free Software Foundation; either
// version 3 of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along with this
// program; if not, see <https://www.gnu.org/licenses/>.
//
// All names, logos, and references to "Deltares" are registered trademarks of Stichting
// Deltares and remain full property of Stichting Deltares at all times. All rights reserved.

using DiKErnel.KernelWrapper.Json.Output.Data.Definitions;
using Newtonsoft.Json;

namespace DiKErnel.KernelWrapper.Json.Output.Data.Generic
{
    internal class JsonOutputLocationData
    {
        public JsonOutputLocationData(JsonOutputFailureLocationData failureData,
                                      JsonOutputDamageLocationData damageData,
                                      JsonOutputPhysicsLocationData physicsData)
        {
            FailureData = failureData;
            DamageData = damageData;
            PhysicsData = physicsData;
        }

        [JsonProperty(JsonOutputDefinitions.ID, NullValueHandling = NullValueHandling.Ignore)]
        public int? Id { get; }

        [JsonProperty(JsonOutputDefinitions.FAILURE)]
        public JsonOutputFailureLocationData FailureData { get; }

        [JsonProperty(JsonOutputDefinitions.DAMAGE, NullValueHandling = NullValueHandling.Ignore)]
        public JsonOutputDamageLocationData DamageData { get; }

        [JsonProperty(JsonOutputDefinitions.PHYSICS, NullValueHandling = NullValueHandling.Ignore)]
        public JsonOutputPhysicsLocationData PhysicsData { get; }
    }
}